odoo/models.py

```diff
@@ -5851,6 +5851,33 @@ def filtered(self, func):
             self.mapped(name)
         return self.browse([rec.id for rec in self if func(rec)])
 
+    def _group_on(self, key):
+        """Eagerly groups the records of ``self`` by the ``key``, returning a
+        dict from the ``key``'s result to recordsets. All the resulting
+        recordsets are guaranteed to be part of the same prefetch-set.
+
+        Provides a convenience method to partition existing recordsets without
+        the overhead of a :meth:`~.read_group`, but performs no aggregation.
+
+        .. note:: unlike :func:`itertools.groupby`, does not care about input
+                  ordering, however the tradeoff is that it can not be lazy
```

Comment:
valuable note 💯 

most of the time, I see the mistake of using `itertools.groupby` without sorting first, causing unexpected result